compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -909,6 +953,142 @@ impl<'interner> Monomorphizer<'interner> {
+            self.debug_types.insert_var(fe_var_id, &var_def.name, var_type)
+        } else {
+            self.debug_types.get_var_id(fe_var_id).unwrap()
+        };
+        let interned_var_id = self.intern_var_id(var_id, &call.location);
+        arguments[0] = self.expr(interned_var_id);
+    }
+
+    /// Update instrumentation code for a variable being dropped out of scope.
+    /// Given the fe_var_id we search for the last assigned runtime variable ID
+    /// and replace it instead.
+    fn patch_debug_var_drop(
+        &mut self,
+        call: &HirCallExpression,
+        arguments: &mut [ast::Expression],
+    ) {
+        let hir_arguments = vecmap(&call.arguments, |id| self.interner.expression(id));
+        let Some(HirExpression::Literal(HirLiteral::Integer(fe_var_id, _))) = hir_arguments.get(0) else {
+            unreachable!("Missing FE var ID in __debug_var_drop call");
+        };
+        // update variable drops (ie. when the var goes out of scope)
+        let fe_var_id = fe_var_id.to_u128() as u32;
+        if let Some(var_id) = self.debug_types.get_var_id(fe_var_id) {
+            let interned_var_id = self.intern_var_id(var_id, &call.location);
+            arguments[0] = self.expr(interned_var_id);
+        }
+    }
+
+    /// Update instrumentation code inserted when assigning to a member of an
+    /// existing variable. Same as above for replacing the fe_var_id, but also
+    /// we need to resolve the path and the type of the member being assigned.
+    /// For this last part, we need to resolve the mapping from field names in
+    /// structs to positions in the runtime tuple, since all structs are
+    /// replaced by tuples during compilation.
+    fn patch_debug_member_assign(
+        &mut self,
+        call: &HirCallExpression,
+        arguments: &mut [ast::Expression],
+        arity: usize,
+    ) {
+        let hir_arguments = vecmap(&call.arguments, |id| self.interner.expression(id));
+        let Some(HirExpression::Literal(HirLiteral::Integer(fe_var_id, _))) = hir_arguments.get(0) else {
+            unreachable!("Missing FE var ID in __debug_member_assign call");
+        };
+        let Some(HirExpression::Ident(HirIdent { id, .. })) = hir_arguments.get(1) else {
+            unreachable!("Missing value identifier in __debug_member_assign call");
+        };
+        // update variable member assignments
+        let var_def_name = self.interner.definition(*id).name.clone();
+        let var_type = self.interner.id_type(call.arguments[2]);
+        let fe_var_id = fe_var_id.to_u128() as u32;
+
+        let mut cursor_type = self
+            .debug_types
+            .get_type(fe_var_id)
+            .unwrap_or_else(|| panic!("type not found for fe_var_id={fe_var_id}"))
+            .clone();
+        for i in 0..arity {
+            if let Some(HirExpression::Literal(HirLiteral::Integer(fe_i, i_neg))) =
+                hir_arguments.get(2 + i)
```

Comment:
2?
Just parsing this quickly I'm unsure what the offset is for. Can you use a named constant instead?

Reply:
I introduced constants to name the argument slots for the debug oracle functions. 2 was the indeed the position of the first index to reference member assignments (right after the var_id and the updated value for the member).